org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/launcher/SourceElementQualifierProvider.java

```diff
@@ -49,25 +46,18 @@ public String getText(Object element) {
 		if (element instanceof IJavaElement) {
 			IJavaElement parent = ((IJavaElement)element).getParent();
 			return fJavaLabels.getText(parent);
-		} else if (element instanceof ZipEntryStorage) {
-			ZipEntryStorage storage = (ZipEntryStorage)element;
-			try (ZipFile archive = storage.getArchive()) {
-				String zipFileName = archive.getName();
-				IPath path = new Path(zipFileName);
-				IRuntimeClasspathEntry entry = JavaRuntime.newArchiveRuntimeClasspathEntry(path);
-				IResource res = entry.getResource();
-				if (res == null) {
-					// external
-					return zipFileName;
-				}
-				// internal
-				return res.getName();
-			} catch (IOException e) {
-				e.printStackTrace();
+		} else if (element instanceof ZipEntryStorage storage) {
+			@SuppressWarnings("resource") // Don't close the archive to not break subsequent usages
+			String zipFileName = storage.getArchive().getName();
+			IPath path = IPath.fromOSString(zipFileName);
```

Comment:
With the new API introduce in https://github.com/eclipse-platform/eclipse.platform/pull/1545, this could avoid the call to `getArchive()` and would avoid suppressing the warning.
Therefore I suggest we await that.